### DISCUSSION TITLE
[node] does not export `TestContext` type

### DISCUSSION THREAD

**Discussion by khaosdoctor** [DefinitelyTyped/DefinitelyTyped#69488] [Issues with a @types package]
I don't know if this has been raised before, or if it's by design. But when we're doing tests using `node:test`, it's possible to use the `TestContext` type within a test like so:

```ts
//                                      \/ this 'ctx' is of type "TestContext"
it('should something', (ctx) => {})
```

However, this type is never exported in the Node types, and it could be super useful especially for mocking, since global mocking using `mock.fn()` can conflict with other tests as they're not reset after each test. So, let's say you have a mock factory like:

```ts
function mockFactory(data: someType, t: TestContext) {
	return t.mock.fn(...)
}
```

It would be extremely useful to have this type in there. Another example [can be found here](https://github.com/Formacao-Typescript/projeto-3/blob/706edc628eba7e38696d83659790f7e43e6350b1/src/services/ClassService.test.ts#L44-L45) in a test repository I'm doing with my students (this is where we picked this). The change seems to be easy as the class is already in the `test.d.ts` file line 386. Adding `export type TestContext = TestContext` would solve it:

```ts
    class TestContext {
       // the definition
    }
    export type TestContext = TestContext; // << here
```

Wanted to discuss this with the community if you think it would be a good addition.

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment]
TestContext was already added, it can be closed.
